topic_resources/ada_cooks.md

```diff
@@ -1,71 +1,88 @@
-- Extra credit for additional methods to add neat functionality
-- Extra credit for writing additional specs
-- Extra credit for attractive or inventive user interface
-
+### As any User, I can...
+- search by the name of an Ingredient or Gadget and get related Recipes back.
+- __Recipes__
+  - view an alphabetical list of all Recipes
+  - view the details of a Recipe, inluding...
+    - a link to the User profile of the person who created the Recipe
+    - a link to the Gadget profile for any associated Gadgets
+    - links to the Ingridient profile for all associated Ingredients
+- __Ingredients__
+  - view an alphabetical list of all Ingedients
+  - view the details of an individual Ingredient, including...
+    - a list of links to Recipes associated with the Ingredient
+- __Gadgets__
+  - view an alphabetical list of all Gadgets
+  - view the details of one Gadget, including...
+    - a list of links to Recipes associated with the Gadget
+    
+### As an anonymous User I can...
+- register a new User account
+- Log In to an existing User account
+
+### As a Logged In User I can...
+- Log Out
+- __Cookbooks__
+  - See a list of Cookbooks I've already created, including...
+    - The number of Recipes associated with the Cookbook
+    - The number of unique Gadgets associated with the Cookbook
+    - The number of unique Recipes associated with the Cookbook
+  - See the details of a specific Cookbook, including...
+    - a list of links to the Recipes associated with the Cookbook
+    - a link to remove (unassociate) a Recipe from the Cookbook
+  - Create a new Cookbook, associated with my User account, with...
+    - a `name` (required)
+    - a `description` (optional)
+  - Destroy an existing Cookbook
+    - Destroying a Cookbook __does not__ destroy the Recipes, just the associations.
+  - Edit an existing Cookbook's `name` and `description` information.
+- __Recipes__
+  - Create a new Recipe, optionally associated with one of my existing Cookbooks, with...
+    - a `name` (required)
+    - a `description` (optional)
+    - an uploaded image of the prepared Recipe (optional)
+    - a `preparation` (required)
+    - one or more Ingredients (required)
+    - zero or more Gadgets (optional)
+  - Edit an existing Recipe that I created by...
+    - Changing the `name`, `description`, uploaded image, and/or `preparation`
+    - Removing or Adding Ingredients
+    - Removing or Adding Gadgets
+  - Destroy an existing Recipe that I created
+    - Destroying a Recipe __does not__ destroy the Ingrdients or Gadgets, just the associations
+  - From a Recipe's detail page, I can save the Recipe to an existing Cookbook
+- __Ingredients__
+  - Create an Ingredient with...
+    - a unique `name` (required)
+    - an uploaded image of the Ingredient (optional)
+  - Edit the `name` and/or the uploaded image of an existing Ingredient that I created
+  - Destroy an existing Ingredient that I created
+- __Gadgets__
```

Comment:
I'm totally fine with removing it completely. I think with the emphasis on testing, advanced AR stuffs, and introducing file handling, there's plenty of meat here.